crates/core/src/client/errors.rs

```diff
@@ -92,12 +91,21 @@ impl<T: std::error::Error> From<EthApiError<T>> for ErrorObject<'static> {
     }
 }
 
-impl<T: std::error::Error> From<EthApiError<T>> for jsonrpsee::core::Error {
-    fn from(err: EthApiError<T>) -> Self {
+impl<E: std::error::Error> From<EthApiError<E>> for jsonrpsee::core::Error {
+    fn from(err: EthApiError<E>) -> Self {
         jsonrpsee::core::Error::Call(err.into())
     }
 }
 
+/// Converting directly from `ConversionError` to `jsonrpsee::core::Error` in case conversions are
+/// done in eth-rpc crate.
+/// TODO: Remove this when conversions are done in core crate only.
+impl From<ConversionError> for ErrorObject<'static> {
```

Comment:
I think we can remove this and just use `into()` to go from ConversionError to EthApiError. The idea of conversion error is that it should always live inside a EthApiError, hence the from implementation.

Reply:
alright!